### DISCUSSION TITLE
Migrations not tracked by git

### DISCUSSION THREAD

**Discussion by Grafcube** [diesel-rs/diesel#3521] [Q&A]
I'm fairly new to this so I might be misunderstanding something. When I use Diesel CLI, it creates the `migrations/` directory with a `.keep` file. But after I run `diesel migration generate <something>` git still only tracks `migrations/.keep` and completely ignores everything else in that directory. I even checked my gitignore but I can't seem to figure out why this is happening.

Any help would be appreciated.

**Comment by weiznich** [comment]
I'm not sure whether I can follow your question. Diesel does not track anything in your git repository for you. You need to explicitly add new migrations via `git add migrations/whatever` as that's the case for any other file in your git repository.

**Reply by Grafcube** [reply]
Sorry I should have been a bit more specific. After doing `git add migrations` here is git status:
![image](https://user-images.githubusercontent.com/34889224/220080132-8fef40cc-fae2-483c-99c6-4f0ce94e3768.png)

And here is the contents of `migrations/`:
![image](https://user-images.githubusercontent.com/34889224/220080279-d3f861ae-dbfe-40c1-aafc-f66468f96346.png)

I am not able to add the contents to git and I don't know why.

**Reply by weiznich** [reply]
I fear that's impossible to answer without access to your local setup. You can be assured that diesel does not set anything that would result these files being not visible for git. You might want to check the content of your `.gitignore` file (also the global ones). Otherwise I would say that we as diesel project cannot provide support on that as that's not an diesel issue.

**Reply by Grafcube** [reply]
I suspected as much but I wasn't sure. I'll see if there's something I missed. Thanks anyway.

Edit: I forgot that I added a global gitignore a long time ago that I copied from somewhere and one of the patterns was `*.sql` :facepalm:
